integration/suite_test.go

```diff
@@ -0,0 +1,191 @@
+		publisherAuthOption: withPublisherEnvManagedIdentityClientID(),
+	}
+	suite.Run(t, clientIdSuite)
+}
+
+func TestResourceID(t *testing.T) {
+	t.Parallel()
+	resourceIdSuite := &serviceBusSuite{
+		Prefix:              "rid-",
+		listenerAuthOption:  withEnvManagedIdentityResourceID(),
+		publisherAuthOption: withPublisherEnvManagedIdentityResourceID(),
+	}
+	suite.Run(t, resourceIdSuite)
+}
+
+func withEnvConnectionString() listener.ManagementOption {
+	connStr := os.Getenv("SERVICEBUS_CONNECTION_STRING") // `Endpoint=sb://XXXX.servicebus.windows.net/;SharedAccessKeyName=XXXX;SharedAccessKey=XXXX`
+	if connStr == "" {
+		panic("environment variable SERVICEBUS_CONNECTION_STRING was not set")
+	}
+	return listener.WithConnectionString(connStr)
+}
+
+func withEnvPublisherConnectionString() publisher.ManagementOption {
+	connStr := os.Getenv("SERVICEBUS_CONNECTION_STRING") // `Endpoint=sb://XXXX.servicebus.windows.net/;SharedAccessKeyName=XXXX;SharedAccessKey=XXXX`
+	if connStr == "" {
+		panic("environment variable SERVICEBUS_CONNECTION_STRING was not set")
+	}
+
+	return publisher.WithConnectionString(connStr)
+}
+
+func withEnvManagedIdentityClientID() listener.ManagementOption {
+	serviceBusNamespaceName := os.Getenv("SERVICEBUS_NAMESPACE_NAME") // `Endpoint=sb://XXXX.servicebus.windows.net/;SharedAccessKeyName=XXXX;SharedAccessKey=XXXX`
+	if serviceBusNamespaceName == "" {
+		panic("environment variable SERVICEBUS_NAMESPACE_NAME was not set")
+	}
+
+	// if managedIdentityClientID is empty then library will assume system assigned managed identity
+	managedIdentityClientID := os.Getenv("MANAGED_IDENTITY_CLIENT_ID")
+	if managedIdentityClientID == "" {
+		panic("environment variable MANAGED_IDENTITY_CLIENT_ID was not set")
+	}
+	provider, err := aad.NewJWTProvider(
+		aad.JWTProviderWithManagedIdentityClientID(managedIdentityClientID, ""),
+		aad.JWTProviderWithResourceURI(serviceBusResourceURI))
+	if err != nil {
+		panic(err.Error())
+	}
+	return listener.WithTokenProvider(serviceBusNamespaceName, provider)
+}
+
+func withEnvManagedIdentityResourceID() listener.ManagementOption {
+	serviceBusNamespaceName := os.Getenv("SERVICEBUS_NAMESPACE_NAME") // `Endpoint=sb://XXXX.servicebus.windows.net/;SharedAccessKeyName=XXXX;SharedAccessKey=XXXX`
+	if serviceBusNamespaceName == "" {
+		panic("environment variable SERVICEBUS_NAMESPACE_NAME was not set")
+	}
+
+	// if managedIdentityClientID is empty then library will assume system assigned managed identity
+	managedIdentityResourceID := os.Getenv("MANAGED_IDENTITY_RESOURCE_ID")
+	if managedIdentityResourceID == "" {
+		panic("environment variable MANAGED_IDENTITY_RESOURCE_ID was not set")
+	}
+
+	provider, err := aad.NewJWTProvider(
+		aad.JWTProviderWithManagedIdentityResourceID(managedIdentityResourceID, ""),
+		aad.JWTProviderWithResourceURI(serviceBusResourceURI))
+	if err != nil {
+		panic(err.Error())
+	}
+	return listener.WithTokenProvider(serviceBusNamespaceName, provider)
```

Comment:
any reason we're preferring `WithTokenProvider` instead of `WithManagedIdentityResourceID` during the test? to make serviceBusResourceURI configurable later?

Reply:
Yes,
WithManagedIsentityResourceID will pass a new token provider for each test.
IMDS barfs with timeout VERY quickly when you send it concurrent requests. I'm going to report it as a bug in fact...
Using TokenProvoder allows to run the 3 suites in parallel